newdle/api.py

```diff
@@ -373,7 +406,15 @@ def get_participants(code):
     )
     if newdle.private and (g.user is None or newdle.creator_uid != g.user['uid']):
         raise Forbidden('You cannot view the participants of this newdle')
-    return RestrictedParticipantSchema(many=True).jsonify(newdle.participants)
+    participants = RestrictedParticipantSchema(many=True).dump(newdle.participants)
+    return jsonify(
+        [
+            sign_user(p, fields={'email', 'name', 'auth_uid'})
+            if p['auth_uid'] is not None
+            else p
+            for p in participants
+        ]
```

Comment:
This might better fit in a `post_dump` hook on the schema itself (or a subclass of it it we use it in other places as well)?

Reply:
Sure, we're using it everywhere so, sounds good.

Reply:
In the same way, we can also add another `post_dump` to `UserSearchResultSchema`.